specifyweb/frontend/js_src/lib/components/WorkBench/WbView.tsx

```diff
@@ -37,25 +37,417 @@ import { hasPermission } from '../Permissions/helpers';
 import { userPreferences } from '../Preferences/userPreferences';
 import type { UploadPlan } from '../WbPlanView/uploadPlanParser';
 import type { Dataset } from '../WbPlanView/Wrapped';
-import { WbCellMeta } from './CellMeta';
+import { WbCellCounts, WbCellMeta, WbCellMetaReact } from './CellMeta';
 import { DataSetNameView } from './DataSetMeta';
 import { DevShowPlan } from './DevShowPlan';
-import { Disambiguation } from './DisambiguationLogic';
+import { Disambiguation, DisambiguationReact } from './DisambiguationLogic';
 import { configureHandsontable, getHotPlugin } from './handsontable';
 import { downloadDataSet } from './helpers';
-import { getHotHooks } from './hooks';
+import { getHotHooks, getHotHooksReact } from './hooks';
 import { getSelectedRegions } from './hotHelpers';
 import type { WbMapping } from './mapping';
 import { parseWbMappings } from './mapping';
 import { fetchWbPickLists } from './pickLists';
 import { WbUploaded } from './Results';
 import { wbViewTemplate } from './Template';
-import { WbActions } from './WbActions';
-import { WbUtils } from './WbUtils';
-import { WbValidation } from './WbValidation';
+import { WbActions, WbActionsComponent } from './WbActions';
+import { WbUtils, WbUtilsComponent, WbUtilsReact } from './WbUtils';
+import { WbValidation, WbValidationReact } from './WbValidation';
+import type { LocalizedString } from 'typesafe-i18n';
+import { className } from '../Atoms/className';
+import { Input } from '../Atoms/Form';
+import { DataSetName } from './DataSetMeta';
+import { WbSpreadsheet } from './WbSpreadsheet';
+import { useBooleanState } from '../../hooks/useBooleanState';
+import { WbToolkit } from './WbToolkit';
+import {
+  WbAdvancedSearch,
+  getInitialSearchPreferences,
+} from './AdvancedSearch';
+import type { WbSearchPreferences } from './AdvancedSearch';
+import { loadingBar } from '../Molecules';
 
 export type WbStatus = 'unupload' | 'upload' | 'validate';
 
+export type Workbench = {
```

Comment:
Do you have any suggestions on how I can go about doing that? I initially tried to decouple things but the classes `Disambiguation`, `CellMeta`, `Validation` all use each others' functions and so I'm finding it hard to decouple them.

Reply:
there may be ways to decouple them, but I would have to study the code a bit deeper to give you good suggestions
if you don't mind, could I do that after you are done with majority of the changes and have most TODOs/comments resolved? or would you prefer that feedback earlier?

Reply:
After looking at this 3 weeks later, the way the code looks now is pretty good - the 4 tightly coupled classes (Utils, Cells, Meta, Disambiguation) don't really do rendering directly, so they are not directly refactorable into components/hooks.

And, it's always easier to write less coupled code than to refactor existing code to be less couples.

But, if you are interested in decoupling these further, here is how I would go about it:

Make a list of what properties from the workbench object what class is using to see the dependency graph:
  - Cells: disambiguation, hot, utils, cellCounts
  - Disambiguation: dataset, data, mappings, hot, validation, cells
  - Validation: dataset, hot, mappings, cells, utils
  - Utils: cells, hot, dataset, mappings

One way to reduce the main `Workbench` object would be to pass hot, dataset, mappings and data to these classes directly as individual arguments rather as part of the `Workbench` object, and then see if one of these 4 classes could be made a child of the other to create a parent->child relationship rather than entangled siblings. For example, Disambiguation could be made a child of Validation

Then, could try to move some of the methods out of the class and turn them into utility functions that receive props. In the process, if you see that some function is mainly called from a different file then maybe you could move that function into the different file

Then, check how many of the methods and properties from these classes you can make private.
For example, see how in just a few lines here, 3 different properties from the utils class are accessed: https://github.com/specify/specify7/blob/07a1e5b0846469d7e4b3068077039eb313a4f40e/specifyweb/frontend/js_src/lib/components/WorkBench/hooks.ts#L267-L278. Can that be replaced by a method on the utils class to hide it's internals a bit more?